specs/IsNonClientRegionSupportEnabled.md

```diff
@@ -0,0 +1,176 @@
+    {
+        coreWebView2Settings.IsNonClientRegionSupportEnabled = false;
+    }
+}
+```
+
+## Declaring Non-client App Regions
+Non-client regions are HTML elements that are marked with the css style `app-region`.
+* Draggable regions can be declared through the values `drag` or `no-drag`. 
+    * `app-region: drag` will support [draggable region functionality](#description) for the html element.
+    * `app-region: no-drag` will change cursor to I-bar, with text highlighting enabled. 
+    Elements with this style will not support draggable region functionality.
+```html
+<!DOCTYPE html>
+<body>
+    <div style="app-region:drag">Drag Region</div>
+    <div style="app-region:no-drag">No-drag Region</div>
+</body>
+<html>
+```
+
+# Remarks
+If the feature flag (`msWebView2EnableDraggableRegions`) is used to enable the CSS style `app-region`'s
+values `drag` and `no-drag`, draggable region support will remain enabled 
+even if the `IsNonClientRegionSupportEnabled` setting is `FALSE`. 
+* Note: The feature flag is experimental and should not be used in production.
+
+# API Notes
+See [API Details](#api-details) section below for API reference.
+
+# API Details
+## Win32 C++
+```cpp
+
+[uuid(436CA5E2-2D50-43C7-9735-E760F299439E), object, pointer_default(unique)]
+interface ICoreWebView2Settings12 : ICoreWebView2Settings11 {
+  /// The `IsNonClientRegionSupportEnabled` property enables web pages to use the 
+  /// `app-region` CSS style. Disabling/Enabling the `IsNonClientRegionSupportEnabled`
+  /// takes effect after the next navigation. Defaults to `FALSE`.
+  /// 
+  /// When this property is `TRUE`, then all the following non-client region support 
+  /// will be enabled:
+  /// 1. Draggable Regions will be enabled and treated as a window's title bar. 
+  /// Draggable Regions are non-client regions on a webpage that are exposed through the css
+  /// attribute `app-region` and can take the values `drag` or `no-drag`. When set to 
+  /// `drag`, these regions will be treated like the window's title bar, supporting 
+  /// dragging of the entire WebView and its host app window, the title bar context menu
+  /// upon right click, and the maximizing to fill the window and restoring the window size
+  /// upon double click of the html element. 
+  ///
+  /// When set to `FALSE`, then all non-client region support will be disabled. Web
+  /// pages will not be able to use the `app-region` CSS style.
+  [propget] HRESULT IsNonClientRegionSupportEnabled([out, retval] BOOL* enabled);
+  /// Set the IsNonClientRegionSupportEnabled property
+  [propput] HRESULT IsNonClientRegionSupportEnabled([in] BOOL enabled);
+}
+```
+
+## .NET and WinRT
+```c#
+namespace Microsoft.Web.WebView2.Core
+{
+    runtimeclass CoreWebView2Settings
+    {
+        [interface_name("Microsoft.Web.WebView2.Core.ICoreWebView2Settings12")]
+        {
+            Boolean IsNonClientRegionSupportEnabled { get; set; };
```

Comment:
The feature flag is named 'msWebView2EnableDraggableRegions', the CSS style is 'app-region' and this setting calls it 'NonClientRegion'. We also had a different feature recently that gave this a name although I don't know what we settled on. Is 'NonClientRegion' the name we want to use going forward?

@bradp0721

Reply:
I believe that NonClientRegion is the name that we are using moving forward. The recent feature which I believe you are referring to is the one for enabling draggable regions for Visual Hosting. The implementation also uses the name NonClientRegion

Reply:
Yes, we had discussed this as a team and landed on NonClientRegion instead of AppRegion, as it more closely aligns with the windowing concepts that it aims to emulate.